instrumentation-genai/opentelemetry-instrumentation-vertexai/src/opentelemetry/instrumentation/vertexai/utils.py

```diff
@@ -0,0 +1,122 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import re
+from dataclasses import dataclass
+from os import environ
+from typing import (
+    TYPE_CHECKING,
+    Mapping,
+    Optional,
+    Sequence,
+)
+
+from opentelemetry.semconv._incubating.attributes import (
+    gen_ai_attributes as GenAIAttributes,
+)
+from opentelemetry.util.types import AttributeValue
+
+if TYPE_CHECKING:
+    from google.cloud.aiplatform_v1.types import content, tool
+
+
+@dataclass(frozen=True)
+class GenerateContentParams:
+    model: str
+    contents: Optional[Sequence[content.Content]] = None
+    system_instruction: Optional[content.Content | None] = None
+    tools: Optional[Sequence[tool.Tool]] = None
+    tool_config: Optional[tool.ToolConfig] = None
+    labels: Optional[Mapping[str, str]] = None
+    safety_settings: Optional[Sequence[content.SafetySetting]] = None
+    generation_config: Optional[content.GenerationConfig] = None
+
+
+def get_genai_request_attributes(
+    params: GenerateContentParams,
+    operation_name: GenAIAttributes.GenAiOperationNameValues = GenAIAttributes.GenAiOperationNameValues.CHAT,
+):
+    model = _get_model_name(params.model)
+    generation_config = params.generation_config
+    attributes: dict[str, AttributeValue] = {
```

Comment:
You are not using Attributes because of the Optional right? We should probably drop it.

Reply:
Ya that's right. What do you mean by drop it though?

Reply:
Drop Optional from Attributes definition

Reply:
I think that would be a breaking change in our typing though. Most of the API allows you to pass None anywhere attributes is: https://github.com/open-telemetry/opentelemetry-python/blob/58f2d161d4b772ce6e62d9f40ba9de16445f4193/opentelemetry-api/src/opentelemetry/trace/__init__.py#L291

We could add another alias for this